packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/mappers/ReactTextMapper.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.mappers
+
+import android.widget.TextView
+import androidx.annotation.VisibleForTesting
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.sessionreplay.internal.AsyncJobStatusCallback
+import com.datadog.android.sessionreplay.internal.recorder.MappingContext
+import com.datadog.android.sessionreplay.internal.recorder.mapper.TextViewMapper
+import com.datadog.android.sessionreplay.model.MobileSegment
+import com.datadog.reactnative.sessionreplay.NoopTextPropertiesResolver
+import com.datadog.reactnative.sessionreplay.ReactTextPropertiesResolver
+import com.datadog.reactnative.sessionreplay.TextPropertiesResolver
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.uimanager.UIManagerModule
+
+internal class ReactTextMapper(): TextViewMapper() {
+    private var reactTextPropertiesResolver: TextPropertiesResolver =
+        NoopTextPropertiesResolver()
+
+    internal constructor(
+        logger: InternalLogger,
+        reactContext: ReactContext,
+        uiManagerModule: UIManagerModule?
+    ): this() {
+        if (uiManagerModule == null) {
+            this.reactTextPropertiesResolver = NoopTextPropertiesResolver()
+        } else {
+            this.reactTextPropertiesResolver = ReactTextPropertiesResolver(
+                logger = logger,
+                reactContext = reactContext,
+                uiManagerModule = uiManagerModule
+            )
+        }
+    }
```

Comment:
avoid having `var` for no reason:

```suggestion
internal class ReactTextMapper(private val reactTextPropertiesResolver: TextPropertiesResolver): TextViewMapper() {

    internal constructor(
        logger: InternalLogger,
        reactContext: ReactContext,
        uiManagerModule: UIManagerModule?
    ): this(
        reactTextPropertiesResolver = if (uiManagerModule == null) {
            NoopTextPropertiesResolver()
        } else {
            ReactTextPropertiesResolver(
                logger = logger,
                reactContext = reactContext,
                uiManagerModule = uiManagerModule
            )
        }
    )
```